lib/mayaUsd/fileio/utils/adaptor.cpp

```diff
@@ -838,6 +857,32 @@ UsdMayaSchemaAdaptor::CreateAttribute(const TfToken& attrName, MDGModifier& modi
         // Maya, because it won't behave like the fallback value in USD.)
         UsdMayaReadUtil::SetMayaAttr(plug, attrDef->GetDefaultValue(), modifier);
     }
+#else
+    std::string                  mayaAttrName = _GetMayaAttrNameOrAlias(attrName);
+    std::string                  mayaNiceAttrName = attrDef.GetName();
+    MFnDependencyNode            node(_handle.object());
+
+    bool    newAttr = !node.hasAttribute(mayaAttrName.c_str());
+    MObject attrObj = UsdMayaReadUtil::FindOrCreateMayaAttr(
+        attrDef.GetTypeName(),
+        attrDef.GetVariability(),
+        node,
+        mayaAttrName,
+        mayaNiceAttrName,
+        modifier);
+    if (attrObj.isNull()) {
+        return UsdMayaAttributeAdaptor();
+    }
+
+    MPlug   plug = node.findPlug(attrObj);
```

Comment:
AM I wrong that the difference actually only start from here? I think the ifdef/else could be made shorter. Maybe a helper function could make the code identical.

Reply:
It's subtly different... The old `attrDef` was a handle so we had to using the de-referencing operator when calling into its functions. The new one is a structure so we no longer dereference